src/settings/groups/GroupDetailsModel.ts

```diff
@@ -0,0 +1,318 @@
+	groupInfo: GroupInfo
+	private readonly group: LazyLoaded<Group>
+	private usedStorageInBytes!: number
+	private readonly members: LazyLoaded<Array<GroupInfo>>
+	private administratedGroups!: LazyLoaded<Array<GroupInfo>>
+	private localAdminGroupInfo: LazyLoaded<Array<GroupInfo>>
+
+	constructor(
+		groupInfo: GroupInfo,
+		private readonly entityClient: EntityClient,
+		private readonly updateViewCallback: () => void,
+	) {
+		this.entityClient = entityClient
+		this.groupInfo = groupInfo
+		this.group = new LazyLoaded(() => this.entityClient.load(GroupTypeRef, this.groupInfo.group))
+
+		this.group.getAsync().then(() => this.updateViewCallback())
+
+		this.members = new LazyLoaded(async () => {
+			const group = await this.group.getAsync()
+			// load only up to 200 members to avoid too long loading, like for account groups
+			const groupMembers = await this.entityClient.loadRange(GroupMemberTypeRef, group.members, GENERATED_MIN_ID, 200, false)
+			return promiseMap(groupMembers, member => this.entityClient.load(GroupInfoTypeRef, member.userGroupInfo))
+		})
+
+		// noinspection JSIgnoredPromiseFromCall
+		this.updateMembers()
+
+		if (this.groupInfo.groupType === GroupType.LocalAdmin) {
+			this.administratedGroups = new LazyLoaded(async () => {
+				const group = await this.group.getAsync()
+				// load only up to 200 members to avoid too long loading, like for account groups
+				const administratedGroups = await this.entityClient.loadRange(
+					AdministratedGroupTypeRef,
+					assertNotNull(group.administratedGroups).items,
+					GENERATED_MAX_ID,
+					200,
+					true
+				)
+				return promiseMap(administratedGroups, administratedGroup => this.entityClient.load(GroupInfoTypeRef, administratedGroup.groupInfo))
+			})
+
+			// noinspection JSIgnoredPromiseFromCall
+			this.updateAdministratedGroups()
+		}
+
+		this.localAdminGroupInfo = new LazyLoaded(() => localAdminGroupInfoModel.init())
+
+		this.localAdminGroupInfo.getAsync().then(() => this.updateViewCallback())
+
+		// noinspection JSIgnoredPromiseFromCall
+		this.updateUsedStorage()
+	}
+
+	isGroupActive(): boolean {
+		return this.groupInfo.deleted == null
+	}
+
+	getGroupType(): string | null {
+		return this.group.isLoaded()
+			? this.group.getLoaded().type
+			: null
+
+	}
+
+	getGroupName(): string {
+		return this.groupInfo.name
+	}
+
+	getUsedStorage(): number | null {
+		return this.usedStorageInBytes
+	}
+
+	getCreationDate(): Date {
+		return this.groupInfo.created
+	}
+
+	getMembersInfo(): Array<GroupInfo> {
+		return this.members.isLoaded()
+			? this.members.getLoaded()
+			: []
+	}
+
+	getAdministratedGroups(): Array<GroupInfo> {
+		return this.administratedGroups.isLoaded()
+			? this.administratedGroups.getLoaded()
+			: []
+	}
+
+	getGroupMailAddress(): string {
+		return this.groupInfo.mailAddress ?? ""
+	}
+
+	/**
+	 * remove the group of the given groupInfo from this group
+	 */
+	removeGroupMember(userGroupInfo: GroupInfo): void {
+		showProgressDialog(
```

Comment:
showing dialogs from the model is generally eh

Reply:
also do we need to redraw afterwards

Reply:
redraw seems to not be necessary